### ISSUE TITLE
Not calling second rule

### ISSUE THREAD

**Issue by karthik6244** [lutticoelho/RulesChain#2]
In the code, either FakeBirthdayDiscountRule Or MyNameIsLuttiDiscount, only one is getting called. 
Use method is not calling both the rules.

**Comment by lutticoelho** [comment]
Hi @karthik6244. Thank you for the feedback.

I ran the tests again and I wasn't able to reproduce the problem. `RuleChain.Use` method was called twice, one for each Rule added in the chain.

But I believe that maybe your question is about the method `Rule.Run` instead of `RuleChain.Use`.
And yes, only `MyNameIsLuttiDiscount` rule executes the method `Run`, thats beacause of `ShouldRun` method at `FakeBirthdayDiscountRule` will always return false.

Basically the method `Run` will only be executed if same class `ShouldRun` returns true.

If you have any more questions I'll be happy to answer. Feel free to share any specific case that you have in mind.

Best regards

**Comment by karthik6244** [comment]
Hi lutticoelho, I am attaching the screenshot below, When I tried to execute action method About in Home controller, it's throwing MissingMethodException. 
![ErrorScreenshot](https://user-images.githubusercontent.com/36606474/115821671-d71f6600-a420-11eb-8a04-a54999d80a0c.png)

Best Regards,

**Comment by lutticoelho** [comment]
Hi @karthik6244. Sorry about that, but today this project still not support Dependency Injection. I haven't work into it for a long time. I really want to fix it, but it demands a huge refactor on how the `Rule` classes are added into the `RuleChain`.

**Comment by karthik6244** [comment]
Hi lutticoelho, instead of adding the fake rules I am trying to mock the calls, if possible can you post a piece of code mocking one rule.  To Mock one rule I have written like below, 
                var _mockedObject = Substitute.For<Rule<ApplyDiscountContext>>();

                var fakeBirthdayDiscountRule = new FakeBirthdayDiscountRule(_mockedObject);

                var res = fakeBirthdayDiscountRule.Run(context);
                Assert.Equal(context1, res);
But I want multiple rules to be mocked, like chaining of rules. Any suggestions

**Comment by lutticoelho** [comment]
Hi @karthik6244. I realise that we should not need to mock a rule to run unit tests for a single rule. Instead of that, we should use `EndOfChainRule`. 

I've opened a pull request for that. Do you mind in take a look into it?

https://github.com/lutticoelho/RulesChain/pull/3

**Comment by lutticoelho** [comment]
Hi @karthik6244. 

I finished the adjustments to support dependency injection on Rules, updated the [sample project](https://bit.ly/ruleschain-sample) and created a [.Net Fiddle](https://bit.ly/RulesChain-DotNetFiddle) with a sample using dependency injection.

Now are are on [version 1.0.3](http://bit.ly/RulesChain-nugetorg)

If you have any further questions feel free to open another issue.

Best regards.
